### PR TITLE
gnome-terminal: Add cursor and highlight color settings

### PR DESCRIPTION
**Pull request by liff**
[See also](https://gitlab.gnome.org/GNOME/gnome-terminal/blob/3.30.2/src/org.gnome.Terminal.gschema.xml#L156).

### CODE REVIEW

File: modules/programs/gnome-terminal.nix [hunk 1]
```diff
@@ -115,6 +153,11 @@ let
           background-color = pcfg.colors.backgroundColor;
           palette = pcfg.colors.palette;
         }
+        // (
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: The surrounding `(` and `)` seem unnecessary here.

File: modules/programs/gnome-terminal.nix [hunk 2]
```diff
@@ -123,6 +166,20 @@ let
             bold-color = pcfg.colors.boldColor;
           }
         )
+        // (
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: The surrounding `(` and `)` seem unnecessary here.

File: modules/programs/gnome-terminal.nix [hunk 3]
```diff
@@ -123,6 +166,20 @@ let
...
+          optionalAttrs (cursorColorsSet) {
+            cursor-colors-set = true;
+            cursor-foreground-color = pcfg.colors.cursorForegroundColor;
+            cursor-background-color = pcfg.colors.cursorBackgroundColor;
+          }
+        )
+        // (
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: The surrounding `(` and `)` seem unnecessary here.

File: modules/programs/gnome-terminal.nix [hunk 4]
```diff
@@ -30,10 +30,41 @@ let
           description = "The bold color, null to use same as foreground.";
         };
 
+        allowBold = mkOption {
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: I think this belongs outside the `colors` option.

File: modules/programs/gnome-terminal.nix [hunk 5]
```diff
@@ -30,10 +30,41 @@ let
...
+          type = types.nullOr types.bool;
+          description = "If true, allow applications in the terminal to make text boldface.";
+        };
+
         palette = mkOption {
           type = types.listOf types.str;
           description = "The terminal palette.";
         };
+
+        cursorForegroundColor = mkOption {
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: The `Color` in the end can be omitted from this and the below options.

I think the naming of `foregroundColor`, `backgroundColor`, and `boldColor` was a mistake and they should have been named without the `Color`. I might change this in the future.

File: modules/programs/gnome-terminal.nix [hunk 6]
```diff
@@ -30,10 +30,41 @@ let
           description = "The bold color, null to use same as foreground.";
         };
 
+        allowBold = mkOption {
+          default = null;
+          type = types.nullOr types.bool;
+          description = "If true, allow applications in the terminal to make text boldface.";
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: Can break the long descriptions over multiple lines to keep them below 80 chars wide. E.g.
```
          description = ''
            If true, allow applications in the terminal to make text boldface.
          '';
```

Similar holds for the other options.

File: modules/programs/gnome-terminal.nix [hunk 7]
```diff
@@ -30,10 +30,41 @@ let
...
           description = "The terminal palette.";
         };
+
+        cursorForegroundColor = mkOption {
+          default = null;
+          type = types.nullOr types.str;
+          description = "The color for the foreground of the text character at the terminal’s cursor position.";
+        };
+
+        cursorBackgroundColor = mkOption {
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: Could be `cursor.background`?

File: modules/programs/gnome-terminal.nix (hunk 5 above)
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: Could be `cursor.foreground`?

File: modules/programs/gnome-terminal.nix [hunk 8]
```diff
@@ -30,10 +30,41 @@ let
...
+          default = null;
+          type = types.nullOr types.str;
+          description = "The color of the background of the terminal’s cursor.";
+        };
+
+        highlightForegroundColor = mkOption {
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: Could be `highlight.foreground`?

File: modules/programs/gnome-terminal.nix [hunk 9]
```diff
@@ -30,10 +30,41 @@ let
...
+          default = null;
+          type = types.nullOr types.str;
+          description = "The color for the foreground of the text character at the terminal’s highlight position.";
+        };
+
+        highlightBackgroundColor = mkOption {
```
Comment (rycee) [inline modules/programs/gnome-terminal.nix]: Could be `highlight.background`?

**Comment by rycee** [comment]
I think this is a really nice addition and sorry about the many comments! I guess I felt a bit picky tonight 😀

**Comment by liff** [comment]
No need to be sorry, picky is good :smiley:

**Comment by rycee** [comment]
Thanks! Rebased to master in 03162970cd6dadfac58f169cd50aed5a5aeec14f. I made a few minor changes, in particular, if the cursor or highligh colors are not set then include `cursor-colors-set = false` and `highlight-colors-set = false` in the Gnome Terminal configuration. Otherwise it would be impossible to unset previously set colors.

**Pull request closed without merging**